flows/parent_flow.py

```diff
@@ -1,47 +1,367 @@
+import logging
+import os
+import json
 from enum import Enum
 
-from prefect import flow, get_run_logger
+import mlflow
+from mlflow.tracking import MlflowClient
+from prefect import flow, task, get_run_logger
+from prefect.deployments import run_deployment
+from prefect.states import Failed
 
-from flows.conda.conda_flows import launch_conda
-from flows.docker.docker_flows import launch_docker
-from flows.podman.podman_flows import launch_podman
-from flows.slurm.slurm_flows import launch_slurm
+# Import the Prefect client to check flow run states
+from prefect.client import get_client
 
+# Import schema classes for validation
+from flows.conda.schema import CondaParams
+from flows.docker.schema import DockerParams
+from flows.podman.schema import PodmanParams
+from flows.slurm.schema import SlurmParams
+
+logger = logging.getLogger(__name__)
+
+# MLflow connection parameters - load from environment variables
+MLFLOW_TRACKING_URI = os.getenv("MLFLOW_TRACKING_URI", "")
+MLFLOW_TRACKING_USERNAME = os.getenv("MLFLOW_TRACKING_USERNAME", "")
+MLFLOW_TRACKING_PASSWORD = os.getenv("MLFLOW_TRACKING_PASSWORD", "")
 
 class FlowType(str, Enum):
     podman = "podman"
     conda = "conda"
     slurm = "slurm"
     docker = "docker"
 
+@task
+def determine_best_environment(hpc_type: str) -> FlowType:
+    """
+    Determine the best execution environment based on hpc_type.
+    
+    Args:
+        hpc_type: Type of HPC to execute on
+    
+    Returns:
+        Best flow type to use
+    """
+    logger = get_run_logger()
+    
+    # Map HPC type to flow type
+    hpc_type = hpc_type.lower()
+    if hpc_type == "nersc":
+        logger.info(f"HPC type is NERSC, selecting SLURM")
+        return FlowType.slurm
+    elif hpc_type == "nsls-ii":
+        logger.info(f"HPC type is NSLS-II, selecting PODMAN")
+        return FlowType.podman
+    elif hpc_type == "als":
+        logger.info(f"HPC type is ALS cluster-ball, selecting DOCKER")
+        return FlowType.docker
+    elif hpc_type in [ft.value for ft in FlowType]:
+        # If the hpc_type is actually a flow type, use it directly
+        return FlowType(hpc_type)
+    else:
+        # Default to conda for unknown HPC types
+        logger.info(f"Unknown HPC type: {hpc_type}, defaulting to CONDA environment")
+        return FlowType.conda
+
+@task
+def get_algorithm_details_from_mlflow(model_name: str):
```

Comment:
Should we add model version? If not defined, should we go for the latest?
```suggestion
def get_algorithm_details_from_mlflow(model_name: str, model_version: Optional[int] = None)
```

Reply:
The `model_name` sent from the application side follows the same naming convention as in `default_model.json`:
`"model_name": "pytorch_autoencoder_v0.0.5"`
where the version is included. It is also consistent with the original algorithm display:
<img width="450" height="106" alt="image" src="https://github.com/user-attachments/assets/440e4db3-9d9d-4b57-8c4a-1ea391e37c61" />

Alternatively, I could split it into two fields—`model_name`: "pytorch_autoencoder" and `model_version`: "1" (instead of "v0.0.5")—and have the application send them as separate variables. Note that MLflow assigns numeric versions (1, 2, …, n, with n being the latest) by default. If we want a custom version name (e.g., "v0.0.5"), we need to add it manually as a `tag` (which I have already done). Let me know which approach you think is better.

Reply:
Splitting it into 2 fields and following mlflow version tracking convention sounds like a good plan to me